samcli/commands/init/init_templates.py

```diff
@@ -179,7 +181,7 @@ def get_preprocessed_manifest(
         # at the top of list template example displayed to the Customer.
         preprocessed_manifest = {"Hello World Example": {}}  # type: dict
         for template_runtime in manifest_body:
-            if filter_value and filter_value != template_runtime:
+            if not filter_value_matches_template_runtime(filter_value, template_runtime):
                 continue
```

Comment:
Let's add `debug` logging here so that we can investigate any possible issues in the future.